RadarSDK/RadarRegion.m

```diff
@@ -30,40 +30,57 @@ - (nullable instancetype)initWithObject:(nullable id)object {
         return nil;
     }
     
-    NSDictionary *regionDict = (NSDictionary *)object;
+    NSDictionary *dict = (NSDictionary *)object;
 
     NSString *_id = @"";
     NSString *name = @"";
     NSString *code = @"";
     NSString *type = @"";
     NSString *flag = @"";
     
-    id eventIdObj = regionDict[@"_id"];
-    if ([eventIdObj isKindOfClass:[NSString class]]) {
-        _id = (NSString *)eventIdObj;
+    id idObj = dict[@"_id"];
+    if ([idObj isKindOfClass:[NSString class]]) {
+        _id = (NSString *)idObj;
     }
 
-    id eventNameObj = regionDict[@"name"];
-    if ([eventNameObj isKindOfClass:[NSString class]]) {
-        name = (NSString *)eventNameObj;
+    id nameObj = dict[@"name"];
+    if ([nameObj isKindOfClass:[NSString class]]) {
+        name = (NSString *)nameObj;
     }
     
-    id eventCodeObj = regionDict[@"code"];
-    if ([eventCodeObj isKindOfClass:[NSString class]]) {
-        code = (NSString *)eventCodeObj;
+    id codeObj = dict[@"code"];
+    if ([codeObj isKindOfClass:[NSString class]]) {
+        code = (NSString *)codeObj;
     }
     
-    id eventTypeObj = regionDict[@"type"];
-    if ([eventTypeObj isKindOfClass:[NSString class]]) {
-        type = (NSString *)eventTypeObj;
+    id typeObj = dict[@"type"];
+    if ([typeObj isKindOfClass:[NSString class]]) {
+        type = (NSString *)typeObj;
     }
 
-    id flagObj = regionDict[@"flag"];
+    id flagObj = dict[@"flag"];
     if (flagObj && [flagObj isKindOfClass:[NSString class]]) {
         flag = (NSString *)flagObj;
     }
-    
-    return [[RadarRegion alloc] initWithId:_id name:name code:code type:type flag:flag];
+        
+    if (_id && name && code && type) {
+        return [[RadarRegion alloc] initWithId:_id name:name code:code type:type flag:flag];
+    }
+
+    return nil;
+}
+
```

Comment:
nit: extra `\n`?

Reply:
Love the attention to detail. For real. Fixing